scripts/apps/search/components/fields/AssociatedItemsList.tsx

```diff
@@ -14,32 +14,50 @@ interface IProps {
 }
 
 interface IState {
-    associations: Array<IArticle>;
+    types: Array<string>;
+}
+
+interface IRelated {
+    _id: string;
+    type: string;
 }
 
 /**
  * Render icon with count for each item type present in associations.
  */
 export class AssociatedItemsList extends React.Component<IProps, IState> {
-    readonly state = {associations: []};
+    readonly state = {types: []};
 
     componentDidMount() {
         this.getAssociations();
     }
 
     getAssociations() {
         const {content} = this.props.svc;
+        const associations = this.props.item.associations || {};
+        const related = Object.values(associations)
+            .filter((_related) => _related != null);
+        const relatedTypes = related
+            .map((_related: IRelated) => _related.type)
+            .filter((type) => type != null);
+
+        if (relatedTypes.length === related.length) { // types for all
```

Comment:
What is this checking? If all related items have types? Why does it matter?

Reply:
because that field displays how many items are there for each type. so it checks if all associations have types. it's for BC

Reply:
What kind of backward compatibility are you referring to? The one where associations might have an `_id`, but not `type` in which case you fetch it from a server separately? Can you add a comment to the code about this as well?

It's not easy to understand this as is, after a few months I'll be clueless why the code is the way it is.

Reply:
When you write an upgrade script related to `_internal` property in another PR, It'd be useful if you could write an upgrade script for this as well and then remove the condition you added for compatibility.

Reply:
yep makes sense